[macOS] Compile errors when building with C#

**Godot version:**

Godot 3.2.x from commit 7b4b83e9dcb91b44b39e8cc87df3ddecdb29b689

**OS/device including version:**

macOS 10.14.6

**Issue description:**

Trying to follow [the instructions to build with C# support](https://docs.godotengine.org/en/stable/development/compiling/compiling_with_mono.html#doc-compiling-with-mono).

I'm using the following options to scons:

```
target=debug -j 8 dev=yes use_llvm=yes tools=yes module_mono_enabled=yes mono_glue=no
```

This produces many build errors because `-Werror` and `-Winconsistent-missing-override` are both specified.

Example build errors:

```
In file included from modules/mono/csharp_script.cpp:31:
modules/mono/csharp_script.h:70:2: error: 'get_class' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
        GDCLASS(CSharpScript, Script);
        ^
./core/object.h:271:17: note: expanded from macro 'GDCLASS'
        virtual String get_class() const {                                                                                                  \
                       ^
./core/script_language.h:101:2: note: overridden virtual function is here
        GDCLASS(Script, Resource);
        ^
./core/object.h:271:17: note: expanded from macro 'GDCLASS'
        virtual String get_class() const {                                                                                                  \
                       ^
In file included from modules/mono/csharp_script.cpp:31:
modules/mono/csharp_script.h:70:2: error: '_get_class_namev' overrides a member function but is not marked 'override' [-Werror,-Winconsistent-missing-override]
        GDCLASS(CSharpScript, Script);
        ^
./core/object.h:274:28: note: expanded from macro 'GDCLASS'
        virtual const StringName *_get_class_namev() const {                                                                                \
                                  ^
./core/script_language.h:101:2: note: overridden virtual function is here
        GDCLASS(Script, Resource);
        ^
./core/object.h:274:28: note: expanded from macro 'GDCLASS'
        virtual const StringName *_get_class_namev() const {                                                                                \
                                  ^
```

<details>
<summary>Example compile command</summary>

clang++ -o modules/mono/mono_gd/gd_mono_method.osx.tools.64.o -c -std=gnu++14 -Wctor-dtor-privacy -Wnon-virtual-dtor -g3 -arch x86_64 -mmacosx-version-min=10.9 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Werror -DDEBUG_ENABLED -DDEBUG_MEMORY_ENABLED -DDEBUG_MEMORY_ALLOC -DDISABLE_FORCED_INLINE -DTYPED_METHOD_BIND -DOSX_ENABLED -DUNIX_ENABLED -DGLES_ENABLED -DAPPLE_STYLE_KEYS -DCOREAUDIO_ENABLED -DCOREMIDI_ENABLED -DGL_SILENCE_DEPRECATION -DPTRCALL_ENABLED -DTOOLS_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -DGLAD_ENABLED -DGLES_OVER_GL -DMODULE_ASSIMP_ENABLED -DMODULE_BMP_ENABLED -DMODULE_BULLET_ENABLED -DMODULE_CAMERA_ENABLED -DMODULE_CSG_ENABLED -DMODULE_CVTT_ENABLED -DMODULE_DDS_ENABLED -DMODULE_ENET_ENABLED -DMODULE_ETC_ENABLED -DMODULE_FREETYPE_ENABLED -DMODULE_GDNATIVE_ENABLED -DMODULE_GDSCRIPT_ENABLED -DMODULE_GRIDMAP_ENABLED -DMODULE_HDR_ENABLED -DMODULE_JPG_ENABLED -DMODULE_JSONRPC_ENABLED -DMODULE_MBEDTLS_ENABLED -DMODULE_MOBILE_VR_ENABLED -DMODULE_MONO_ENABLED -DHAVE_CXX11_THREAD_LOCAL -DGD_MONO_HOT_RELOAD -D_REENTRANT -I/Library/Frameworks/Mono.framework/Versions/Current/include/mono-2.0 -Ithirdparty/libpng -Ithirdparty/glad -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/osx -I. modules/mono/mono_gd/gd_mono_method.cpp
</details>

--- a/modules/mono/csharp_script.h
+++ b/modules/mono/csharp_script.h
@@ -179,7 +179,7 @@ public:
 	virtual void get_script_property_list(List<PropertyInfo> *p_list) const;
 	virtual void update_exports();
 
-	void get_members(Set<StringName> *p_members) override;
+	virtual void get_members(Set<StringName> *p_members);
 
 	virtual bool is_tool() const { return tool; }
 	virtual bool is_valid() const { return valid; }

